### PR TITLE
chore: Add issue template for Refactoring tickets

### LINKED ISSUE #2980: 🚀 [Feature] Add issue template for Refactors
### User Problem

I find myself needing to choose between `Bug` and `Feature`, when sometimes I want to create a `Refactor` ticket. 
### Implementation

We can add another template for `Refactor`

### PR DESCRIPTION
**Pull request by mattwr18**
- We want to have an issue created for each PR merged in, sometimes we
  are just refactoring deprecated or legacy code and there should be an
option to create these tickets.

### Issues

- fixes #2980 

### Todo

- [X] None

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/refactor_tickets.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+name: 🐛 Refactor ticket
+about: Help us improve our code by refactoring it.
+labels: refactor
+title: 🐛 [Refactor] 
+---
+
+## :zap: Refactor ticket
+<!-- Describe your issue in detail. Include screenshots if needed. Give us as much information as possible. Use a clear and concise description of what the bug is.-->
```
Comment (roschaefer) [inline .github/ISSUE_TEMPLATE/refactor_tickets.md]:
```suggestion
<!-- Describe your issue in detail. Include screenshots if needed. Give us as much information as possible. Use a clear and concise description of what the problem is.-->
```

File: .github/ISSUE_TEMPLATE/refactor_tickets.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Motive
+<!-- What is the purpose of this refactoring? If it's removing depcrecated code, please link to the deprecation notice. -->
+
+### Implementation
+<!-- Please, docment any ideas of how the code should be refactored. -->
```
Comment (roschaefer) [inline .github/ISSUE_TEMPLATE/refactor_tickets.md]:
```suggestion
<!-- Please, document any ideas of how the code should be refactored. -->
```

File: .github/ISSUE_TEMPLATE/refactor_tickets.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+name: 🐛 Refactor ticket
```
Comment (alina-beck) [inline .github/ISSUE_TEMPLATE/refactor_tickets.md]:
```suggestion
name: 🔧 Refactor ticket
```

may I suggest a different icon, to make it easier to distinguish between the issue types? :wink:

File: .github/ISSUE_TEMPLATE/refactor_tickets.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+about: Help us improve our code by refactoring it.
+labels: refactor
+title: 🐛 [Refactor]
```
Comment (alina-beck) [inline .github/ISSUE_TEMPLATE/refactor_tickets.md]:
```suggestion
title: 🔧 [Refactor] 
```

same here – feel free to choose another one, of course! it was just the first one that came to mind...

File: .github/ISSUE_TEMPLATE/refactor_tickets.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+labels: refactor
+title: 🐛 [Refactor] 
+---
+
+## :zap: Refactor ticket
+<!-- Describe your issue in detail. Include screenshots if needed. Give us as much information as possible. Use a clear and concise description of what the bug is.-->
+
+### Motive
+<!-- What is the purpose of this refactoring? If it's removing depcrecated code, please link to the deprecation notice. -->
+
```
Comment (alina-beck) [inline .github/ISSUE_TEMPLATE/refactor_tickets.md]: Would a section `Related issues` be helpful, to point out where this came up? Or is it too much? I usually add it under `Additional context` when its relevant but know that not everyone does. :smile:

I'm ok with it either way.

**Pull request merged**